src/sagemaker/tensorflow/deploying_tensorflow_serving.rst

```diff
@@ -269,6 +269,194 @@ More information on how to create ``export_outputs`` can be found in `specifying
+       Args:
+           data (obj): the TensorFlow serving response
+           context (Context): an object containing request and configuration details
+       Returns:
+           (bytes, string): data to return to client, response content type
+       """
+       if data.status_code != 200:
+           raise ValueError(data.content.decode('utf-8'))
+
+       response_content_type = context.accept_header
+       prediction = data.content
+       return prediction, response_content_type
+
+There are occasions when you might want to have complete controls over
+request. For example, making TFS request (REST or GRPC) to first model,
+inspecting results and making request to a second model. In this case,
+you may implement the ``handler`` instead of the ``input_handler`` and
+``output_handler`` pair:
+
+.. code::
+
+   import json
+   import requests
+
+
+   def handler(data, context):
+       """Handle request.
+       Args:
+           data (obj): the request data
+           context (Context): an object containing request and configuration details
+       Returns:
+           (bytes, string): data to return to client, (optional) response content type
+       """
+       processed_input = _process_input(data, context)
+       response = requests.post(context.rest_uri, data=processed_input)
+       return _process_output(response, context)
+
+
+   def _process_input(data, context):
+       if context.request_content_type == 'application/json':
+           # pass through json (assumes it's correctly formed)
+           d = data.read().decode('utf-8')
+           return d if len(d) else ''
+
+       if context.request_content_type == 'text/csv':
+           # very simple csv handler
+           return json.dumps({
+               'instances': [float(x) for x in data.read().decode('utf-8').split(',')]
+           })
+
+       raise ValueError('{{"error": "unsupported content type {}"}}'.format(
+           context.request_content_type or "unknown"))
+
+
+   def _process_output(data, context):
+       if data.status_code != 200:
+           raise ValueError(data.content.decode('utf-8'))
+
+       response_content_type = context.accept_header
+       prediction = data.content
+       return prediction, response_content_type
+
+You can also bring in external dependencies to help with your data
+processing. There are 2 ways to do this:
+
+1. If you included ``requirements.txt`` in your ``source_dir`` or in
+    your dependencies, the container will install the Python dependencies at runtime using ``pip install -r``:
+
+.. code::
+
+    from sagemaker.tensorflow.serving import Model
+
+    model = Model(entry_point='inference.py',
+                  dependencies=['requirements.txt'],
+                  model_data='s3://mybucket/model.tar.gz',
+                  role='MySageMakerRole')
+
+
+2. If you are working in a network-isolation situation or if you don't
+   want to install dependencies at runtime everytime your Endpoint
```

Comment:
"...every time your endpoint starts or a batch transform job runs, you might want to put..."